hibernate-reactive-core/src/test/java/org/hibernate/reactive/SchemaUpdateSqlServerTestBase.java

```diff
@@ -0,0 +1,340 @@
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.SQLSERVER;
+import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.GROUPED;
+import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.INDIVIDUALLY;
+
+public abstract class SchemaUpdateSqlServerTestBase extends BaseReactiveTest {
+
+	public static class IndividuallySchemaUpdateSqlServerTestBase
+			extends SchemaUpdateSqlServerTestBase {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, INDIVIDUALLY.toString() );
+			return configuration;
+		}
+	}
+
+	public static class GroupedSchemaUpdateSqlServerTestBase extends SchemaUpdateSqlServerTestBase {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, GROUPED.toString() );
+			return configuration;
+		}
+	}
+
+	protected Configuration constructConfiguration(String hbm2DdlOption) {
+		Configuration configuration = constructConfiguration();
+		configuration.setProperty( Settings.HBM2DDL_AUTO, hbm2DdlOption );
+		// DEFAULT_CATALOG needs to be set to something other than an empty string;
+		// otherwise, NullPointerException is thrown.
+		configuration.setProperty( Settings.DEFAULT_CATALOG, "master" );
+		configuration.setProperty( Settings.DEFAULT_SCHEMA, "dbo" );
+		configuration.setProperty( Settings.SHOW_SQL, "true" );
+		return configuration;
+	}
+
+	@Rule
+	public DatabaseSelectionRule dbRule = DatabaseSelectionRule.runOnlyFor( SQLSERVER );
+
+	@Before
+	@Override
+	public void before(TestContext context) {
+		Configuration createHbm2ddlConf = constructConfiguration( "create" );
+		createHbm2ddlConf.addAnnotatedClass( ASimpleFirst.class );
+		createHbm2ddlConf.addAnnotatedClass( AOther.class );
+
+		test( context, setupSessionFactory( createHbm2ddlConf )
+				.thenCompose( v -> factoryManager.stop() ) );
+	}
+
+	@After
+	@Override
+	public void after(TestContext context) {
+		final Configuration dropHbm2ddlConf = constructConfiguration( "drop" );
+		dropHbm2ddlConf.addAnnotatedClass( ASimpleNext.class );
+		dropHbm2ddlConf.addAnnotatedClass( AOther.class );
+		dropHbm2ddlConf.addAnnotatedClass( AAnother.class );
+
+		test( context, factoryManager.stop()
+				.thenCompose( v -> setupSessionFactory( dropHbm2ddlConf ) )
+				.thenCompose( v -> factoryManager.stop() ) );
+	}
+
+	@Test
+	public void testValidationSucceed(TestContext context) {
+		Configuration createHbm2ddlConf = constructConfiguration( "validate" );
```

Comment:
I guess this should be callled `validateHbm2ddlConf`

Reply:
I would prefer the generic `configuration`, as used in the other tests.

WDYT?

Reply:
I find `validateHbm2ddlConf` clearer but `configuration` works as well.

I'm happy as long as it is not `createHbm2ddlConf`.

Reply:
I prefer more generic names when it's unambiguous, to avoid future copy/paste errors.

Anyhow, I've fixed it in https://github.com/hibernate/hibernate-reactive/pull/970